ext/dom/namednodemap.c

```diff
@@ -66,43 +54,9 @@ zend_result dom_namednodemap_length_read(dom_object *obj, zval *retval)
 
 /* }}} */
 
-xmlNodePtr php_dom_named_node_map_get_named_item(dom_nnodemap_object *objmap, const zend_string *named, bool may_transform)
-{
-	xmlNodePtr itemnode = NULL;
-	if (objmap != NULL) {
-		if ((objmap->nodetype == XML_NOTATION_NODE) ||
-			objmap->nodetype == XML_ENTITY_NODE) {
-			if (objmap->ht) {
-				if (objmap->nodetype == XML_ENTITY_NODE) {
-					itemnode = (xmlNodePtr)xmlHashLookup(objmap->ht, BAD_CAST ZSTR_VAL(named));
-				} else {
-					xmlNotationPtr notep = xmlHashLookup(objmap->ht, BAD_CAST ZSTR_VAL(named));
-					if (notep) {
-						if (may_transform) {
-							itemnode = create_notation(notep->name, notep->PublicID, notep->SystemID);
-						} else {
-							itemnode = (xmlNodePtr) notep;
-						}
-					}
-				}
-			}
-		} else {
-			xmlNodePtr nodep = dom_object_get_node(objmap->baseobj);
-			if (nodep) {
-				if (php_dom_follow_spec_intern(objmap->baseobj)) {
-					itemnode = (xmlNodePtr) php_dom_get_attribute_node(nodep, BAD_CAST ZSTR_VAL(named), ZSTR_LEN(named));
-				} else {
-					itemnode = (xmlNodePtr) xmlHasProp(nodep, BAD_CAST ZSTR_VAL(named));
-				}
-			}
-		}
-	}
-	return itemnode;
-}
-
-void php_dom_named_node_map_get_named_item_into_zval(dom_nnodemap_object *objmap, const zend_string *named, zval *return_value)
+void php_dom_named_node_map_get_named_item_into_zval(dom_nnodemap_object *objmap, const zend_string *named, const char *ns, zval *return_value)
```

Comment:
~~Does it make sense as a follow-up refactoring to have the namespace be a zend_string instead of a char pointer?~~ Got my answer does not make sense as this is just passed down to libxml.